pkg/render/logstorage/esgateway/esgateway.go

```diff
@@ -0,0 +1,324 @@
+package esgateway
+
+import (
+	"fmt"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/render"
+	relasticsearch "github.com/tigera/operator/pkg/render/common/elasticsearch"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/secret"
+)
+
+const (
+	DeploymentName          = "tigera-es-gateway"
+	ServiceAccountName      = "tigera-es-gateway"
+	RoleName                = "tigera-es-gateway"
+	VolumeName              = "tigera-es-gateway-certs"
+	ElasticsearchUserSecret = "tigera-es-gateway-elasticsearch-access"
+	ServiceName             = "tigera-secure-es-gateway-http"
+	ElasticsearchPortName   = "es-gateway-elasticsearch-port"
+	KibanaPortName          = "es-gateway-kibana-port"
+	Port                    = 5554
+
+	ElasticsearchHTTPSEndpoint = "https://tigera-secure-es-http.tigera-elasticsearch.svc:9200"
+	ElasticsearchPort          = 9200
+	KibanaHTTPSEndpoint        = "https://tigera-secure-kb-http.tigera-kibana.svc:5601"
+	KibanaPort                 = 5601
+)
+
+func EsGateway(
+	logStorage *operatorv1.LogStorage,
+	installation *operatorv1.InstallationSpec,
+	pullSecrets []*corev1.Secret,
+	certSecrets []*corev1.Secret,
+	esAdminUserSecret *corev1.Secret,
+	kibanaInternalCertSecret *corev1.Secret,
+	esInternalCertSecret *corev1.Secret,
+	clusterDomain string,
+) render.Component {
+	secrets := certSecrets
+	tlsAnnotations := map[string]string{}
+
+	// tigera-es-gateway-elasticsearch-access
+	if esAdminUserSecret != nil {
+		elasticsearchUserSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      ElasticsearchUserSecret,
+				Namespace: render.ElasticsearchNamespace,
+			},
+			Data: map[string][]byte{
+				"username": []byte("elastic"),
+				"password": esAdminUserSecret.Data["elastic"],
```

Comment:
I think this same secret is renamed a few times, maybe you should just create a single copy of the admin secret in the Elasticsearch namespace and anything that needs that secret can just mount that.

Reply:
Looks like this was missed after some changes to es-gateway no longer require it to have access to admin credentials. We will now only need one secret with these credentials (for kube-controllers in the elasticsearch namespace).